net/portmapper/portmapper.go

```diff
@@ -337,12 +342,18 @@ func (c *Client) createMapping() {
 	}
 }
 
+// wildcardIP is used when the previous external IP is not known for PCP port mapping.
+var wildcardIP = netaddr.MustParseIP("0.0.0.0")
+
 // createOrGetMapping either creates a new mapping or returns a cached
 // valid one.
 //
 // If no mapping is available, the error will be of type
 // NoMappingError; see IsNoMappingError.
 func (c *Client) createOrGetMapping(ctx context.Context) (external netaddr.IPPort, err error) {
+	if DisableUPnP && DisablePCP && DisablePMP {
+		return
```

Comment:
don't use naked returns on functions over 2 or 3 lines

also, why isn't this `return netaddr.IPPort{}, NoMappingError{ErrGatewayRange}` like line 359? Docs do say:

```
// If no mapping is available, the error will be of type
// NoMappingError; see IsNoMappingError.
```

Reply:
Since all of them are disabled it feels odd to me that it's a `NoMappingError`, but certainly it's more right than returning nil